### PR TITLE
fastapi orjson

### PR DESCRIPTION
**Pull request by raphaelauv**
fastapi doc explicit to use ORJSON to return json whithout check

### CODE REVIEW

File: asgi_bench/runner.py
Comment (provinzkraut) [inline asgi_bench/runner.py:1]: This seems to be unrelated to this PR

**Review by provinzkraut — COMMENTED** [review]
So, the fact that we're not using orjson with FastAPI is very much on purpose.

The idea behind that is simply that all frameworks have so many tweaks and optimisations you can apply, that it would never be a "fair" result. So we just leave everything with its stock configuration.

**Comment by provinzkraut** [comment]
> then we are not comparing apples to apples
> 
> the json endpoint of starlite do not perform any data-validation where fastapi do
> 
> if you think that ORJSONRESPONSE is too optimzed then let's at least use `JSONResponse`

I'm not saying it's too optimised. The point is that it's not the default. FastAPI made the design decision to have this validation by default, and I want to stick to the defaults.

FastAPI also chose to use Pydantic for serialisation and validation, while Litestar uses msgspec, but also supports Pydantic. Does this mean we should explicitly use Pydantic for the Litestar tests so the comparison is more fair?

In my opinion, this doesn't make a lot of sense, since what we're trying to benchmark here isn't "pure framework overhead, if you apply all possible optimisations that suit a particular framework best and get the stack as similar as possible", but simply the frameworks as they come and are going to be used by default.

What I'm trying to say is: If you start deviating from the defaults, it's basically impossible to have a fair comparison because you can't reasonably draw a line at which you could say "this is too much optimisation". We make it very clear that what's being compared here are the frameworks with their "factory default", which I think is a reasonable baseline.

**Comment by raphaelauv** [comment]
when you do not use a XXXResponse object , fastapi will perform data-validation 

so all the endpoints are currently validated and not directly serialize in json , this not an optimization , it's just doing the same than what do starlite 

so currently, here the json fastapi endpoint are using pydantic v1

**Comment by provinzkraut** [comment]
> so all the endpoints are currently validated and not directly serialize in json , this not an optimization , it's just doing the same than what do starlite

FastAPI, by default, applies validation. This is an advertised feature of FastAPI. Turning it off is an optimisation.
Again, the point here is **to compare the frameworks with their default settings**.

> so currently, here the json fastapi endpoint are using pydantic v1

We'll need to upgrade the whole thing to `fastapi>0.100`. PRs for that are welcome.

**Comment by raphaelauv** [comment]
```
FastAPI, by default, applies validation. This is an advertised feature of FastAPI. Turning it off is an optimisation.
Again, the point here is to compare the frameworks with their default settings.
```

so let's add a disclaimer in the benchmark about the JSON case because I sincerely think that `honesty is the best policy`

**Comment by provinzkraut** [comment]
> so let's add a disclaimer in the benchmark about the JSON case because I sincerely think that `honesty is the best policy`

I'm just trying to understand your reasoning here. I do agree that honesty is good, and we are definitely not trying to be dishonest here. Giving FastAPI an artificial edge would do that tough in my opinion. 

Let me give you a different example. I tested many different webserver configurations with this setup, and most of them favoured one framework more than the others. Sanic for example seems to benefit from stock uvicorn more than uvicorn with uvloop, and blacksheep does not benefit from uvicorn's Cython compiled dependencies nearly as much as all other frameworks.


What's the solution for this then? Of course we could run every framework with the configuration that gives it the best result but then we wouldn't be comparing the same things. 

I am fairly certain that we could design every single benchmark for every single framework in a way that would have it outperform all the others, with the exception of blacksheep because it's just in a league of its own. But what does that tell us?
I think it tells us that it's near impossible to come up with an unbiased and objective benchmark. That's why I chose to use the most common deployment configuration, and strictly stick to the frameworks defaults in every case. I also added [this disclaimer](https://docs.litestar.dev/latest/benchmarks.html#interpreting-the-results) to the benchmarking page, telling people to approach these results with caution, this it's almost guaranteed that they won't translate 1:1 into the real world.

Do I think these benchmarks are perfect? No, I don't. But I do think they are actually more objective by making the deliberate decision to not apply any sort of optimisation and only use the canonical way of doing things in all cases.
